File: python3/maximum_length_of_repeated_subarray.py
# Space : O(nm)
# Time  : O(nm)

class Solution:
    def findLength(self, nums1: List[int], nums2: List[int]) -> int:
        n, m = len(nums1), len(nums2)
        dp = [[0 for _ in range(m+1)] for _ in range(n+1)]

        for i in range(n):
            for j in range(m):
                if nums1[i] == nums2[j]:
                    dp[i + 1][j + 1] = dp[i][j] + 1

        return max(max(x) for x in dp)

File: golang/maximum_length_of_repeated_subarray.go
package main

// Space : O(nm)
// Time  : O(nm)

func findLength(nums1 []int, nums2 []int) int {
    n, m := len(nums1), len(nums2)
	ans := 0
    var dp [][]int
    
    for i := 0; i <= n; i++ {
        var temp []int
        for j := 0; j <= m; j++ {
            temp = append(temp, 0)
        }
        dp = append(dp, temp)
    }
    
    for i := 0; i < n; i++ {
        for j := 0; j < m; j++ {
            if nums1[i] == nums2[j] {
                dp[i+1][j+1] = dp[i][j] + 1
            }
        }
    }
    
    for i := 0; i <= n; i++ {
        for j := 0; j <= m; j++ {
            ans = Max(ans, dp[i][j])
        }
    }
    
    return ans
}